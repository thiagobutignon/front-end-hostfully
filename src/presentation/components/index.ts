export { default as ErrorComponent } from './error/error'
export { default as HeaderComponent } from './header/header'
export { default as InputError } from './input/input-error'
export { default as InputWithLabel } from './input/input-with-label'
export { default as LayoutComponent } from './layout/layout'
export { default as PropertyInfoComponent } from './property-info/property-info'
export { default as PropertyCardComponent } from './select/property-select'
export { default as ThemeModeComponent } from './theme-mode/theme-mode'
