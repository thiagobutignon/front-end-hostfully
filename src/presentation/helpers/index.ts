export * from './populate-form-fields'
export * from './wrap-chakra'
