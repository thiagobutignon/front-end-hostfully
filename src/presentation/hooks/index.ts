export * from './booking'
export * from './error'
export * from './property'
