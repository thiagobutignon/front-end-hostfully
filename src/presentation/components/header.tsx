import { Box, Flex, Menu, Stack, Text } from '@chakra-ui/react'

import { ThemeModeComponent } from '@/presentation/components'

const HeaderComponent: React.FC = () => {
  return (
    <>
      <Box bg="reverseBackground" px={4}>
        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
          <Text color="reverseText" fontSize="xl" fontWeight={700}>
            Hostfully
          </Text>

          <Flex alignItems={'center'}>
            <Stack direction={'row'}>
              <Menu>
                <ThemeModeComponent />
              </Menu>
            </Stack>
          </Flex>
        </Flex>
      </Box>
    </>
  )
}

export default HeaderComponent
