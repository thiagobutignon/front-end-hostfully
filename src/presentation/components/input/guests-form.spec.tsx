import { RenderResult, fireEvent, render } from '@testing-library/react'

import { GuestsFormComponent } from '@/presentation/components'

jest.mock('@/presentation/styles/themes', () => ({
  useReverseColor: () => ({ text: 'black' })
}))

describe('GuestsFormComponent', () => {
  let mockOnGuestInfoChange: any
  let sut: RenderResult

  beforeEach(() => {
    mockOnGuestInfoChange = jest.fn()
    sut = render(
      <GuestsFormComponent
        guests={{
          numberOfGuests: 2,
          guests: [
            { name: '', email: '' },
            { name: '', email: '' }
          ]
        }}
        onGuestInfoChange={mockOnGuestInfoChange}
      />
    )
  })

  it('renders correctly with initial state', () => {
    const { getAllByTestId } = sut

    expect(getAllByTestId('input-email-0')[0]).toBeInTheDocument()
    expect(getAllByTestId('input-name-0')[0]).toBeInTheDocument()
  })

  it('calls onGuestInfoChange with correct values on input change', () => {
    const { getAllByTestId } = sut

    const nameInput = getAllByTestId('input-name-0')[0]
    fireEvent.change(nameInput, { target: { value: 'John Doe' } })

    expect(mockOnGuestInfoChange).toHaveBeenNthCalledWith(1, 0, {
      name: 'John Doe',
      email: ''
    })

    const emailInput = getAllByTestId('input-email-0')[0]
    fireEvent.change(emailInput, { target: { value: 'johndoe@email.com' } })

    expect(mockOnGuestInfoChange).toHaveBeenNthCalledWith(2, 0, {
      name: 'John Doe',
      email: 'johndoe@email.com'
    })
  })

  test('matches snapshot', () => {
    const { asFragment } = sut

    expect(asFragment()).toMatchSnapshot()
  })
})
