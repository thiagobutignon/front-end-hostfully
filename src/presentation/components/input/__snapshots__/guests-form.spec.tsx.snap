// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestsFormComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1xb2ccj"
  >
    <input
      class="chakra-input css-werrn"
      data-testid="input-name-0"
      id="name-0"
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      class="chakra-input css-1lm3etc"
      data-testid="input-email-0"
      id="email-0"
      placeholder="Email"
      type="email"
      value=""
    />
    <input
      class="chakra-input css-werrn"
      data-testid="input-name-1"
      id="name-1"
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      class="chakra-input css-1lm3etc"
      data-testid="input-email-1"
      id="email-1"
      placeholder="Email"
      type="email"
      value=""
    />
  </div>
</DocumentFragment>
`;
